### PR TITLE
ghcjs compat

### PR DESCRIPTION
**Pull request by angerman**
This adds the code from https://github.com/luite/plutus/commit/8a2a90719d07693681835cf2b8fa57d48913748b
incrementally as I'm trying to get this built with nix.

### CODE REVIEW

**Comment by angerman** [comment]
@luite is the original author of this code.

**Comment by angerman** [comment]
Since #1793 was merged, we should be able to drop the whole ghc stub part.

File: language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/GenericSemigroup.hs [hunk 1]
```diff
@@ -1,13 +1,22 @@
+{-# LANGUAGE CPP                  #-}
 {-# LANGUAGE FlexibleContexts     #-}
 {-# LANGUAGE TypeOperators        #-}
 {-# LANGUAGE UndecidableInstances #-}
 
-module Language.PlutusCore.Evaluation.Machine.GenericSemigroup where
+module Language.PlutusCore.Evaluation.Machine.GenericSemigroup (GenericSemigroupMonoid(..)) where
+
+-- TODO when we need to support semigroup >=0.19 only, delete this file
```
Comment (reactormonk) [inline language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/GenericSemigroup.hs]: Just delete this as soon as the stackage snapshot includes the right semigroups. CPP is of the devil (and breaks most tooling).

Comment (angerman) [reply language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/GenericSemigroup.hs]: Can't we just extra-dep a newer semigroup?

Comment (reactormonk) [reply language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/GenericSemigroup.hs]: I tried, it follows with an endless string of dependency mismatches. Maybe time to try again? Newest matching LTS would be 15.0, which seems to be a bit new. https://www.stackage.org/package/semigroups/snapshots

Comment (angerman) [reply language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/GenericSemigroup.hs]: Yea, LTS-15 isn't supported yet due to the shenanigans stackage pulled with the pantry switch. :-/

**Comment by michaelpj** [comment]
I think this is now defunct, especially since we're taking the approach of just disabling the plugin when cross-compiling.

**Pull request closed without merging**